sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/SqlDelightFile.kt

```diff
@@ -74,6 +76,23 @@ abstract class SqlDelightFile(
       .reduce { totalScope, directoryScope -> totalScope.union(directoryScope) }
   }
 
+  private val systemTables get() = dialect.predefinedSystemSchema.map { predefinedSql ->
```

Comment:
this is going to get called a lot since its during schema generation, so we'll have multiple of these files lying around.

I wonder if we could instead have them available on the dialect, so theres a single set of these files that exist per dialect. They could be lazily instantiated to retain the creation flow, but that way they're only ever made once per dialect, and the creation flow for them is on the dialect instead of in the file. You could even change the existing `predefinedSystemSchema` API to be a function that returns the files directly instead of a list of text

Reply:
Oh, that's a good idea, moving the PsiFileFactory into the dialect with `lazy`. I just want to keep the String input, so other developers can easily change/add/contribute the schema using strings.

Reply:
that makes sense, maybe time to change SqliteDialect to an abstract class so it can have a public API exposing the files, but a protected API for implementations that is just strings